app/src/actions/queryActions.js

```diff
@@ -1,16 +1,42 @@
 import * as types from './actionTypes';
 import {apiService} from "../services/apiService";
+import * as notifyActions from './notifyActions';
+import history from '../history';
+
+const QUERY_KEYWORD_SERVICE_PATH = '/query';
+const QUERY_QSL_SERVICE_PATH = '/qsl';
+const QUERY_PARAM_NAME = 'keyword';
+const QSL_PARAM_NAME = 'qslstring';
 
 export const changeQuery = str => ({
   type: types.CHANGE_QUERY,
   query: str
 });
 
-export const submitQuery = () => ({type: types.SUBMIT_QUERY});
+export function submitQuery(query) {
+  return dispatch => {
+    if(query !== '' && query.length >= 3) {
+      history.push('/results?query=' + encodeURIComponent(query));
+    } else {
+      const msg = 'Minimum length of Search word must be 3 characters.';
+      dispatch(notifyActions.showNotify(msg));
+      return;
+    }
+  };
+}
 
 export function fetchQuery(query) {
   return dispatch => {
-    return apiService.getKeyword(query)
+
+    let requestPromise;
+
+    if (query.includes('@')) {
```

Comment:
wouldn't hurt to add a comment here about WHY '@' has something to do with QSL

also '@' might be better as a constant, as well